app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,105 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication::AuthnAzure
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+      dependencies: {
+        role_class: ::Role,
+        resource_class: ::Resource,
+        application_identity_class: ApplicationIdentity,
+        logger: Rails.logger
+      },
+      inputs: %i(account service_id username xms_mirid oid)
```

Comment:
let's change `oid` to `oid_token_field` so it's clear what it is. same goes for `xms_mirid`